src/Util/Image.h

```diff
@@ -61,63 +62,158 @@
 // AxisRange(0, 1) defines the axis range including [0, 1] in this example
 // AxisRange(0, 2) defines the axis range including [0, 1, 2] in this example
 
+/**
+ * @brief Represents a range along an axis.
+ *
+ * This struct stores a range with `from` and `to` values, providing
+ * constructors for different initialization scenarios and utility
+ * functions for comparison and range checking.
+ */
 struct AxisRange {
-    int from, to;
+    int from; ///< The start of the range.
+    int to;   ///< The end of the range.
+
+    /**
+     * @brief Default constructor initialising the range from 0 to ALL_Z.
+     */
     AxisRange() {
         from = 0;
         to = ALL_Z;
     }
+
+    /**
+     * @brief Constructs a range with the same start and end value.
+     *
+     * @param from_and_to_ The value for both `from` and `to`.
+     */
     AxisRange(int from_and_to_) {
         from = to = from_and_to_;
     }
+
+    /**
+     * @brief Constructs a range with a specified start and end.
+     *
+     * @param from_ The start of the range.
+     * @param to_ The end of the range.
+     */
     AxisRange(int from_, int to_) {
         from = from_;
         to = to_;
     }
+
+    /**
+     * @brief Checks if two AxisRange objects are equal.
+     *
+     * Two AxisRange objects are considered equal if both their `from`
+     * and `to` values match.
+     *
+     * @param rhs The other AxisRange to compare with.
```

Comment:
I suggest just `the other AxisRange` or `the AxisRange to compare` or `the reference AxisRange`